torch/csrc/jit/runtime/register_prim_ops.cpp

```diff
@@ -493,6 +493,10 @@ static const std::vector<OperatorGeneratorArgs> opGenArgs{
         TORCH_SELECTIVE_SCHEMA("prim::device(Tensor a) -> Device"),
         device,
         aliasAnalysisFromSchema()),
+    OperatorGeneratorArgs(
+        TORCH_SELECTIVE_SCHEMA("prim::device(Tensor a, int index) -> Device"),
```

Comment:
actually, I think you might want to instead duplicate the `aten::device` op around line ~2200

Reply:
Ouch, I've added a variant for `prim::device`, not `aten::device`. Thanks for catching that
That even changes things, because `aten::device`'s inputs will be `str, int`, not `Tensor, int`